src/UriContextCollection.php

```diff
@@ -154,7 +156,8 @@ public function getAutoRouteByLocale($locale)
     {
         foreach ($this->uriContexts as $uriContext) {
             $autoRoute = $uriContext->getAutoRoute();
-            if ($locale === $autoRoute->getLocale()) {
+
+            if (null !== $autoRoute && $locale === $uriContext->getLocale()) {
```

Comment:
I should have mentioned this test revealed a bug: if the `getAutoRouteByLocale` method is called, when the locale is compared using a context which does not contains any auto route, a PHP error occurs (because of the method call on a null object).